trio/tests/test_sync.py

```diff
@@ -512,6 +510,29 @@
         assert (await q.get()) == 2
 
 
+async def test_Queue_unbuffered():
```

Comment:
For a bit more checking of unbuffered `Queue`, I'd suggest adding a `QueueLock1()` to the `lock_factories` and `lock_factory_names` lists in this file.

Reply:
I don't think this works. Attempting acquire the lock (i.e. `await queue.put()`) will block indefinitely. I definitely like idea to test `Queue(0)` as used for explicit synchronization though, so I may add a test that relies on rendezvous behaviour.

Reply:
> Attempting acquire the lock (i.e. `await queue.put()`) will block indefinitely

Oh duh, of course you're right.